### ISSUE TITLE
Syslog breaks on long *program-name*s

### ISSUE THREAD

**Issue by notmgsk** [rigetti/quilc#240]
```
/Users/mark.skilbeck/hackery/lisp/quilc/superlongnamethatshouldbreakthingshopefully/quilc -R
+-----------------+
|  W E L C O M E  |
|   T O   T H E   |
|  R I G E T T I  |
|     Q U I L     |
| C O M P I L E R |
+-----------------+
Copyright (c) 2016-2019 Rigetti Computing.

! ! ! Error: Malformed input for RFC 5424 syslog message. The violated assertion was
(CL-SYSLOG::VALID-APP-NAME-P CL-SYSLOG::APP-NAME)
```

**Comment by ecpeterson** [comment]
Nice 😎

**Comment by notmgsk** [comment]
@ecpeterson should we just clamp `*program-name*` to the last 40 chars?

**Comment by ecpeterson** [comment]
That seems reasonable to me, or something like `path |-> path[0:5] @ "..." @ path[-(n-32):]`. I imagine the start of the path is generally less useful for debugging, but maybe you'd hit some frustrating situation where you wish it would have told you that an initial segment of the path was wildly off.

**Comment by notmgsk** [comment]
I like that idea :)

**Comment by notmgsk** [comment]
Closed by #247
